hal/src/main/native/sim/MockData/EncoderDataInternal.h

```diff
@@ -19,6 +19,9 @@
 namespace hal {
 class EncoderData {
  public:
+  void SetDigitalChannelA(int16_t channel);
+  int16_t GetDigitalChannelA();
```

Comment:
If I understand the question correctly, I would hesitate because the remainder of that class (and, in fact, most of the hal code) does not use the trailing const.  I would think it better, if that semantic is desired, to introduce it as a patch that changes the class as a whole.

Reply:
Member functions that do not mutate the class's internal state (like getters) should be const-qualified. Since nothing overrides that function and it was added in this patch series, there's no reason not to const-qualify it now.

Reply:
Forgive me, I think I've taken offense where none was meant.  And on a trivial matter, what's worse.  But I feel that  "no reason" dismisses my point and belittles it.  Let me restate it to make sure I was clear.

I believe you are proposing that I introduce a relatively obscure semantic into a body of code in an exceptional way, which I believe would obfuscate the code.  (i.e. why on earth is GetDigitalChannelA() the *only* one that is const qualified?  What's so different about it?)

Now I will make whatever change you request.  But I wanted to be sure you heard my concern and that you believe the benefit outweighs my concern.  You can call my concern a 'tiny reason' if it helps :-).

I would also defer to a request on your part that I submit a patch const qualifying all the gettors of the class.

Reply:
I'm sorry if I sound patronizing, but constness is not an obscure semantic; it's a fundamental feature commonly used in idiomatic C++. It's a way to ensure at compile time that a function does not have side effects, which the user of the function can notice by looking at the function signature and rely on. Here's the relevant C++ core guideline: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rconst-fct. Member functions should be const-qualified unless they mutate internal state. We use this all over the place in the rest of WPILib like `double SpeedController::Get() const`. It's enforcing through the compiler what the API writer already intended (the `override` keyword is another example of this).

Of course, I'm assuming the API user is familiar with the feature, but that should be a good assumption since it's a basic feature covered in basically every C++ book I've seen (I use _Programming: Principles and Practice Using C++, 2nd Ed._ to teach my students, and 9.7.4 of the chapter on classes covers this feature). Appending `const` to the function signature is strictly a benefit because it gives the user more information (namely that the function doesn't cause side effects). This doesn't limit where the user can use the function. In fact, it also allows them to use the function in more places like within other const-qualified functions.

Const-qualifying a function is like a stronger variant of marking a Java variable `final`. You don't have to do it, but you should when you can. EncoderDataInternal.h's other functions don't follow this guideline and need to be fixed (but that can happen in a separate PR).